### PR TITLE
Added math equiv exemptions

### LINKED ISSUE #92: Color Jitter Equivalence
We'd like a slightly flexible def of color jitter because PIL uses scaling and TF uses a delta. Same role, just slightly different math.

### LINKED ISSUE #93: Batch wrapping at epoch boundries
Training set size is not always evenly divisible at batch boundries. One nice way to handle this issue is to "wrap" the batches -- mix samples from end of last epoch with samples from start of next epoch. This makes "every epoch" eval timing tricky. As long as we eval after the wrapped batch it is strictly worse from a work perspective. Is this acceptable?

### LINKED ISSUE #94: Batch norm with Padding
Proposal: Allow for padding to be used when computing batch norms.

### LINKED ISSUE #95: Allow Adam implementations to assume that epsilon is zero in some places.
One of our performance experts has requested the ability to treat epsilon as zero in certain places to better accommodate hardware. Given that epsilon is simply a numerical stability factor and in the "ideal mathematical" case it would be arbitrarily small I would propose that allowing this still conforms to the standard of mathematically equivalent to the reference modulo precision constraints.

### PR DESCRIPTION
**Pull request by petermattson**
Fixes #92 -- Color jitter
Fixes #93 -- Batch wrapping
Fixes #94 -- Batchnorm padding
Fixes #95 -- Adam epsilon

### CODE REVIEW

File: training_rules.adoc [hunk 1]
```diff
@@ -256,6 +256,21 @@ CLOSED: The numerical formats fp64, fp32, fp16, and bfloat16 are pre-approved fo
...
+=== Equivalence exceptions
+The CLOSED division allows limited exemptions to mathematical equivalence between implementations for pragmatic purposes, including:
+
+* Different methods can be used to add color jitter as long as the methods are of a similar distribution and magnitude to the reference.
+
+* If test set size is not evenly divisible by batch size, the last batch in an epoch may be composed of the remaining samples in the epoch, may be padded, or may be a mixed batch composed of samples from the end of one epoch and the start of the next. If the mixed batch technique is used, quality for the ending epoch must be evaluated after the mixed batch.
+
+* Values introduced for padding purposes may be reflected in batch norm computations.
+
+* Adam optimizer implementations may use the very small value epsilon to maintain mathematical stability in slightly different ways, provided that methods are reviewed and approved in advance. One such method involves moving epsilon inside the square root in the parameter update equation.
```
Comment (nvpaulius) [inline training_rules.adoc]: My recollection of the working group's conclusion was that we agree with this in spirit, but would like to see the specifics of the proposed change and then add them to the rules or reference. The spirit being that changes to eps handling do not alter its purpose for only preventing division by zero. One concern would be that eps doesn't get altered in a way that changes the training/convergence behavior when the corresponding momentum is very small or zero.

**Review by nvpaulius — APPROVED** [review]
